### PR TITLE
Fix LoS countdowns on speaker deletion

### LINKED ISSUE #2402: Fix LoS - with active structure level countdowns when user is cleared
See OpenSlides/openslides-client#3555 for context

### PR DESCRIPTION
**Pull request by luisa-beerboom**
Closes #2402 

@rrenkert Does this need a migration to remove previous instances of this bug, or is this fine as is?

### CODE REVIEW

**Review by Elblinator — APPROVED** [review]

**Comment by rrenkert** [comment]
I don't think we need a migration here. The error occurs dependent on the speech state that is likely short when it is the "speeking" state.

File: openslides_backend/action/actions/speaker/delete.py [hunk 1]
```diff
@@ -34,3 +40,30 @@ def check_permissions(self, instance: dict[str, Any]) -> None:
             if meeting_user.get("user_id") == self.user_id and not len(restricted):
                 return
         super().check_permissions(instance)
+
+    def get_updated_instances(self, action_data: ActionData) -> ActionData:
+        self.end_time = round(time())
```
Comment (r-peschke) [inline openslides_backend/action/actions/speaker/delete.py:45]: Seems that this set's always the end_time and will overwrite an already closed speaker

Comment (luisa-beerboom) [reply openslides_backend/action/actions/speaker/delete.py:45]: 1. This is a delete function. Any speaker whose id is given will be deleted in the same step.
2. The end_time variable is looked up only in line 67, for the expressed purpose of being able to calculate the correct `remaining_time` when stopping the structure level countdown.
3. This mirrors the functionality in `speaker.end_speech`.

File: openslides_backend/action/actions/speaker/delete.py [hunk 2]
```diff
@@ -34,3 +40,30 @@ def check_permissions(self, instance: dict[str, Any]) -> None:
...
+                "unpause_time",
+                "structure_level_list_of_speakers_id",
+            ],
+        )
+        if (
+            speaker.get("begin_time")
+            and not speaker.get("end_time")
+            and not speaker.get("pause_time")
+        ):
+            self.decrease_structure_level_countdown(self.end_time, speaker)
```
Comment (r-peschke) [inline openslides_backend/action/actions/speaker/delete.py:67]: if the end_time is only looked up here, then you should get it here and use round(time()) as function call parameter. As consequence the changes in lines 43-47 are not necessary.

**Review by r-peschke — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/speaker/delete.py (hunk 2 above)
Comment (luisa-beerboom) [reply openslides_backend/action/actions/speaker/delete.py:67]: Then there may be discrepancies when multiple speakers are stopped at once

Comment (r-peschke) [reply openslides_backend/action/actions/speaker/delete.py:67]: More than 1 speaker talking at the same would be no good idea at all;-)

File: openslides_backend/action/actions/speaker/delete.py (hunk 1 above)
Comment (r-peschke) [reply openslides_backend/action/actions/speaker/delete.py:45]: IMO you don't need to calculate the remaining time, if the end_time was set before.

**Pull request merged**